internal/storage/postgreSQL/treeSQL/tree.go

```diff
@@ -0,0 +1,85 @@
+package treeSQL
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"github.com/green-ecolution/green-ecolution-backend/internal/storage"
+	"github.com/green-ecolution/green-ecolution-backend/internal/storage/postgreSQL/entities/tree"
+)
+
+type TreeRepository struct {
+	db *sql.DB
+}
+
+func NewTreeRepository(db *sql.DB) *TreeRepository {
+	return &TreeRepository{db: db}
+}
+func (r *TreeRepository) Insert(ctx context.Context, data *tree.TreeEntity) error {
```

Comment:
Take the domain entity
```suggestion
func (r *TreeRepository) Insert(ctx context.Context, data *tree.Tree) error {
```